cpp/src/arrow/compute/kernels/scalar_string_test.cc

```diff
@@ -395,6 +395,12 @@ TYPED_TEST(TestStringKernels, AsciiLower) {
                    "[\"aaazzæÆ&\", null, \"\", \"bbb\"]");
 }
 
+TYPED_TEST(TestStringKernels, AsciiSwapCase) {
+  this->CheckUnary("ascii_swapcase", "[]", this->type(), "[]");
+  this->CheckUnary("ascii_swapcase", "[\"aAazZæÆ&\", null, \"\", \"BbB\"]", this->type(),
+                   "[\"AaAZzæÆ&\", null, \"\", \"bBb\"]");
```

Comment:
Can you add characters that don't have a casing here?

Reply:
Sure. Addressed in [8f65c70](https://github.com/apache/arrow/pull/10855/commits/8f65c700ae86d03d594dcf8957f7f17a583b2a97)